frontend/packages/console-app/src/components/volume-group-snapshot/create-volume-group-snapshot.tsx

```diff
@@ -0,0 +1,328 @@
+      isList: true,
+      namespace,
+      selector: labelExpressions.length > 0 ? { matchExpressions: labelExpressions } : undefined,
+    };
+
+    return watch;
+  }, [namespace, labelExpressions]);
+
+  const [data, loaded, loadError] = useK8sWatchResource<PersistentVolumeClaimKind[]>(resourceWatch);
+
+  const [storageClasses, setStorageClasses] = React.useState<StorageClassResourceKind[]>([]);
+  const storageClassWatch = {
+    kind: StorageClassModel.kind,
+    isList: true,
+  };
+
+  const [storageClassData] = useK8sWatchResource<StorageClassResourceKind[]>(storageClassWatch);
+
+  React.useEffect(() => {
+    if (storageClassData) {
+      setStorageClasses(storageClassData);
+    }
+  }, [storageClassData]);
+
+  const getProvisioner = React.useCallback(
+    (storageClassName: string): string | null => {
+      if (!storageClassName || !storageClasses.length) return null;
+
+      const storageClass = storageClasses.find((sc) => sc.metadata.name === storageClassName);
+      return storageClass?.provisioner || null;
+    },
+    [storageClasses],
+  );
+
+  const validatePVCs = React.useCallback(
+    (pvcs: PersistentVolumeClaimKind[]) => {
+      if (!pvcs || pvcs.length <= 1 || labelExpressions.length === 0) {
+        setValidationError(null);
+        return;
+      }
+
+      const provisioners = new Set();
+      pvcs.forEach((pvc) => {
+        const storageClassName = pvc.spec?.storageClassName;
+        if (storageClassName) {
+          const provisioner = getProvisioner(storageClassName);
+          if (provisioner) {
+            provisioners.add(provisioner);
+          }
+        }
+      });
+
+      if (provisioners.size > 1) {
+        setValidationError(
+          t(
+            'console-app~Selected PVCs have different storage provisioners. All PVCs must have StorageClasses with the same provisioner.',
+          ),
+        );
+        return;
+      }
+
+      setValidationError(null);
+    },
+    [getProvisioner, labelExpressions.length, t],
+  );
+
+  React.useEffect(() => {
+    if (loaded && !loadError) {
+      const newPvcObjs = data || [];
+      setPVCObjs(newPvcObjs);
+
+      // Validate PVCs for same provisioner
+      if (newPvcObjs.length > 1) {
+        validatePVCs(newPvcObjs);
+      } else {
+        setValidationError(null);
+      }
+    }
+  }, [data, loaded, loadError, validatePVCs]);
+
+  React.useEffect(() => {
+    if (loaded && data?.length) {
+      const labelsMap: { [key: string]: string[] } = {};
+      data.forEach((pvc) => {
+        if (pvc.metadata?.labels) {
+          Object.entries(pvc.metadata.labels).forEach(([key, value]) => {
+            if (!labelsMap[key]) {
+              labelsMap[key] = [];
+            }
+            if (!labelsMap[key].includes(value)) {
+              labelsMap[key].push(value);
+            }
+          });
+        }
+      });
+      setLabels(labelsMap);
+    }
+  }, [data, loaded]);
```

Comment:
no need for extra `useState`, u can do it during render phase itself... no need to cause an extra re-render...
```suggestion
 const labels = React.useMemo(() => {
    if (loaded && data?.length) {
      const labelsMap: { [key: string]: string[] } = {};
      data.forEach((pvc) => {
        if (pvc.metadata?.labels) {
          Object.entries(pvc.metadata.labels).forEach(([key, value]) => {
            if (!labelsMap[key]) {
              labelsMap[key] = [];
            }
            if (!labelsMap[key].includes(value)) {
              labelsMap[key].push(value);
            }
          });
        }
      });
      return labelsMap;
    }
  }, [data, loaded]);
```